From 72944b9e997845440b87fd416cfa6e41318d61e5 Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Mon, 15 Sep 2025 10:39:34 +0200
Subject: [PATCH] Fix IDL of SVG spec

- `HTMLHyperlinkElementUtils`: https://github.com/w3c/svgwg/issues/312
- Constructor: https://github.com/w3c/svgwg/pull/1010

Also drop `SVGPathElement`, now that SVG Paths is also being crawled, pending
clarification of status between SVG 2 and SVG Paths.
---
 ed/idl/SVG.idl | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/ed/idl/SVG.idl b/ed/idl/SVG.idl
index d046678667..d1346c82c1 100644
--- a/ed/idl/SVG.idl
+++ b/ed/idl/SVG.idl
@@ -314,8 +314,9 @@ interface mixin SVGElementInstance {
   [SameObject] readonly attribute SVGUseElement? correspondingUseElement;
 };
 
-[Constructor(Animation source, Animatable newTarget), Exposed=Window]
+[Exposed=Window]
 interface ShadowAnimation : Animation {
+  constructor(Animation source, Animatable newTarget);
   [SameObject] readonly attribute Animation sourceAnimation;
 };
 
@@ -418,10 +419,6 @@ interface SVGAnimatedPreserveAspectRatio {
   [SameObject] readonly attribute SVGPreserveAspectRatio animVal;
 };
 
-[Exposed=Window]
-interface SVGPathElement : SVGGeometryElement {
-};
-
 [Exposed=Window]
 interface SVGRectElement : SVGGeometryElement {
   [SameObject] readonly attribute SVGAnimatedLength x;
@@ -671,7 +668,20 @@ interface SVGAElement : SVGGraphicsElement {
 };
 
 SVGAElement includes SVGURIReference;
-SVGAElement includes HTMLHyperlinkElementUtils;
+
+// Inline HTMLHyperlinkElementUtils except href, which conflicts.
+partial interface SVGAElement {
+  readonly attribute USVString origin;
+  [CEReactions] attribute USVString protocol;
+  [CEReactions] attribute USVString username;
+  [CEReactions] attribute USVString password;
+  [CEReactions] attribute USVString host;
+  [CEReactions] attribute USVString hostname;
+  [CEReactions] attribute USVString port;
+  [CEReactions] attribute USVString pathname;
+  [CEReactions] attribute USVString search;
+  [CEReactions] attribute USVString hash;
+};
 
 [Exposed=Window]
 interface SVGViewElement : SVGElement {};
-- 
2.51.0

